### PR TITLE
[Explore vis] Histogram viz echarts

### PR DESCRIPTION
**Pull request by ruanyl**
### Description
Add Echarts implementation of histogram

## Changelog

- refactor: echarts implementation of histogram visualization

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

## Summary by CodeRabbit

* **New Features**
  * Added numeric data binning utility with aggregation (SUM, MEAN, MAX, MIN, COUNT)

* **Refactor**
  * Unified bar/histogram series handling for echarts-based visualizations
  * Improved histogram rendering and axis configuration for echarts

* **Tests**
  * Added comprehensive unit tests covering binning behaviors and edge cases

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/plugins/explore/public/components/visualizations/bar/to_expression.ts [hunk 1]
```diff
@@ -924,6 +927,7 @@ export const createDoubleNumericalBarChart = (
       axisColumnMappings: axisColumnMappings ?? {},
     });
 
+    // TODO: check if this is needed
```
Comment (ruanyl) [inline src/plugins/explore/public/components/visualizations/bar/to_expression.ts:930]: @Qxisylolo can you check if this is needed?

File: src/plugins/explore/public/components/visualizations/utils/data_transformation/bin.ts [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+    } else {
+      aggregatedValue = validRecords.length;
+    }
+
+    return [
+      {
+        category: `${min}-${max}`,
+        value: aggregatedValue,
+      },
+    ];
```
[the remark this answers is not in the document]
Comment (ruanyl) [reply src/plugins/explore/public/components/visualizations/utils/data_transformation/bin.ts:158]: @coderabbitai this is fixed now

File: src/plugins/explore/public/components/visualizations/utils/data_transformation/bin.ts
Comment (Hailong-am) [inline src/plugins/explore/public/components/visualizations/utils/data_transformation/bin.ts:1]: instead of using customized bin logic, can we use existing D3's bin algorithm or ` echarts-stat` mentioned here https://apache.github.io/echarts-handbook/en/concepts/data-transform

Comment (ruanyl) [reply src/plugins/explore/public/components/visualizations/utils/data_transformation/bin.ts:1]: That's a good idea, I also did consider `echarts-stat` but that requires to add one extra npm package just for `bin`, we have plan to add more data transformations in the future, I will get back to this when needing more data transform functions

File: src/plugins/explore/public/components/visualizations/bar/to_expression.ts (hunk 1 above)
Comment (Qxisylolo) [reply src/plugins/explore/public/components/visualizations/bar/to_expression.ts:930]: They are not needed here, i already delete them in my local pr, thanks for checking!
